transform/app/jms/connector.py

```diff
@@ -0,0 +1,113 @@
+        logger.error("received an error '%s'", frame.body)
+
+    def on_message(self, frame):
+        action = frame.headers["destination"].split(".")[-1]
+        raw_collection = frame.headers["destination"].split("/")[-1].split(".")[0]
+
+        if raw_collection == "training_resource":
+            collection = "training"
+        elif raw_collection == "interoperability_record":
+            collection = "interoperability guideline"
+        else:
+            collection = raw_collection
+
+        active = json.loads(frame.body)["active"]
+        suspended = json.loads(frame.body)["suspended"]
+        status = json.loads(frame.body)["status"]
+
+        body = (
+            json.loads(frame.body).get("trainingResource")
+            or [json.loads(frame.body).get("interoperabilityRecord")]
+            or None
+        )
+
+        if action == "create":
+            if (
+                active
+                and not suspended
+                and status in ["approved resource", "approved interoperability record"]
+            ):
+                transform_batch.delay(collection, body, full_update=False)
+            else:
+                logger.info(
+                    f"{collection} with id={body['id']} not created."
+                    f"Condition not satisfied: active should be True,"
+                    "suspended should be False, and status should be 'approved resource'. [%s, %s, %s]",
+                    active,
+                    suspended,
+                    status,
+                )
+        elif action == "update":
+            if (
+                active
+                and not suspended
+                and status in ["approved resource", "approved interoperability record"]
+            ):
+                transform_batch.delay(collection, body, full_update=False)
+            else:
+                if check_document_exists(collection, body):
+                    delete_data_by_id.delay(collection, body)
+
+        elif action == "delete":
+            if check_document_exists(collection, body):
+                delete_data_by_id.delay(collection, body)
+        else:
+            logger.warning(f"New {action=} appeared")
+
+
+def default_subscription_condition(connection: stomp.Connection) -> bool:
+    """"""
+    return connection.is_connected()
+
+
+# pylint: disable=too-many-arguments
+def subscribe_to_queue(
+    host: str,
+    port: int,
+    username: str,
+    password: str,
+    topics: List[str],
+    subscription_id: str,
+    ssl: bool = True,
+    _logger: Optional[logging.Logger] = None,
+    subscription_condition: Optional[Callable] = default_subscription_condition,
+) -> None:
+    """ """
+    connection = stomp.Connection([(host, port)])
+    connection.set_listener("", ProviderListener())
+
+    if _logger is None:
+        _logger = logging.getLogger(__name__)
+    try:
+        connection.connect(username=username, password=password, wait=True, ssl=ssl)
```

Comment:
not sure if that will work for ssl (production).
I think we need to "set_ssl" also SS:
https://github.com/cyfronet-fid/eosc-search-service/blob/de4b6be5291d2c14f06b62329916316cda4a9169/backend/app/dependencies/user_actions.py#L34

Or from the script that I have sent you:
```
    c = s.Connection([(JMS_HOST, JMS_PORT)])
    c.set_listener("", MyListener())
    c.set_ssl(for_hosts=[(JMS_HOST, JMS_PORT)], ssl_version=ssl.PROTOCOL_TLS)
    c.connect(JMS_LOGIN, JMS_PASSWORD, wait=True)
```

Reply:
Modified the SSL handling.